app/gui2/src/components/ComponentBrowser/filtering.ts

```diff
@@ -185,18 +265,45 @@ export class Filtering {
   pattern?: FilteringWithPattern
   selfType?: QualifiedName | undefined
   qualifiedName?: FilteringQualifiedName
+  fullPattern: string | undefined
+  /** The first and last match are the parts of the string that are outside of the match.
+   * The middle matches come in groups of three, and contain respectively:
+   * - the unmatched prefix (must end with a `_`)
+   *   (an empty string if the entire qualified name segment was matched)
+   * - the matched text
+   * - the unmatched suffix (an empty string if the entire qualified name segment was matched)
+   * - the separator (`.` or `_`, or the empty string if this is the last segment) */
+  extractMatchesRegex: RegExp | undefined
   showUnstable: boolean = false
   showLocal: boolean = false
 
   constructor(filter: Filter) {
     const { pattern, selfType, qualifiedNamePattern, showUnstable, showLocal } = filter
-    if (pattern != null && pattern !== '') {
+    if (pattern) {
       this.pattern = new FilteringWithPattern(pattern)
     }
     this.selfType = selfType
-    if (qualifiedNamePattern != null && qualifiedNamePattern !== '') {
+    if (qualifiedNamePattern) {
       this.qualifiedName = new FilteringQualifiedName(qualifiedNamePattern)
     }
+    if (qualifiedNamePattern) {
+      this.fullPattern = pattern ? `${qualifiedNamePattern}.${pattern}` : qualifiedNamePattern
+    } else {
+      this.fullPattern = pattern
+    }
+    if (this.fullPattern) {
```

Comment:
From the above code, I don't see any branch where `this.fullPattern` remains undefined.

Reply:
`pattern` may be `undefined`. i'll guard the `else` with an `if` to make it clearer